lib/src/main/java/com/auth0/android/lock/utils/CustomField.java

```diff
@@ -56,22 +58,66 @@
         int TYPE_EMAIL = 3;
     }
 
+    /**
+     * Location to store the field into.
+     */
+    @IntDef({PROFILE_ROOT, USER_METADATA})
+    @Retention(RetentionPolicy.SOURCE)
+    public @interface Storage {
+        /**
+         * Store the field into the user profile root.
+         * The list of attributes that can be added to your root profile is here https://auth0.com/docs/api/management/v2#!/Users/patch_users_by_id.
+         */
+        int PROFILE_ROOT = 0;
+
+        /**
+         * Store the field into the user_metadata object.
+         * The list of attributes that can be added to your root profile is here https://auth0.com/docs/api/management/v2#!/Users/patch_users_by_id.
+         */
+        int USER_METADATA = 1;
+    }
+
     @DrawableRes
-    private int icon;
+    private final int icon;
     @FieldType
     private final int type;
     private final String key;
     @StringRes
     private final int hint;
-
+    @Storage
+    private final int storage;
+
+
+    /**
+     * Constructor for CustomField instance
+     *
+     * @param icon the icon resource to display next to the field.
+     * @param type the type of input this field will receive. Used to determine the applied validation.
+     * @param key  the key to store this field as.
+     * @param hint the placeholder to display when this field is empty.
+     */
     public CustomField(@DrawableRes int icon, @FieldType int type, @NonNull String key, @StringRes int hint) {
```

Comment:
Should I deprecate this existing constructor that doesn't take the `Storage`? I mean.. it still works with the defaults (user metadata) so no real justification.

Reply:
Your decision, personally I think it's fine as is, it's not like there is a requirement for people to move to root storage.